linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/router_clients.js

```diff
@@ -0,0 +1,105 @@
+var RouterClient = (function() {
+  var template;
+
+  function getMetricDefinition(routerName, clientName) {
+    return [
+      {
+        description: "Requests",
+        metricSuffix: "requests",
+        query: Query.clientQuery().withRouter(routerName).withClient(clientName).withMetric("requests").build()
```

Comment:
Disagree that `withFoo` implies copying over mutating, there's plenty of examples the other way, especially when it comes to the builder pattern:
https://www.thekua.com/atwork/2013/04/a-builder-pattern-implementation-in-javascript/
http://brainbaking.com/enhancing-the-builder-pattern-with-closures/

fwiw jQuery is the most well-known builder pattern lib in use for js, and it mutates using `.foo(bar)`